webapp/src/webapp/ai_data_masking/events.cljs

```diff
@@ -40,6 +41,8 @@
  :ai-data-masking->get-by-id-success
  (fn [db [_ data]]
    (-> db
+       ;; Clear any existing connections state to prevent stale data
+       (update-in [:ai-data-masking :active-rule :data] dissoc :connections :connections-loading :connections-error)
        (assoc-in [:ai-data-masking :active-rule :status] :success)
        (assoc-in [:ai-data-masking :active-rule :data] data))))
```

Comment:
Too many iterations in the db

```suggestion
       (update-in db [:ai-data-masking :active-rule :data] merge data)
```

If this `data` doesn't have all keys dissociated, you can pass it with data example:

(merge {:connection nil, :connections-loading false, :connections-error nil} data)